src/Solana/Program.cpp

```diff
@@ -64,20 +65,17 @@ Address TokenProgram::defaultTokenAddress(const Address& mainAddress, const Addr
 Address TokenProgram::findProgramAddress(const std::vector<TW::Data>& seeds, const Address& programId) {
     Address result(Data(32));
     // cycle through seeds for the rare case when result is not valid
-    for (uint8_t seed = 255; seed >= 0; --seed) {
-        std::vector<Data> seedsCopy;
-        for (auto& s: seeds) {
-            seedsCopy.push_back(s);
-        }
-        // add extra seed
-        seedsCopy.push_back({seed});
+    auto bumpSeed = Data{std::numeric_limits<std::uint8_t>::max()};
```

Comment:
I don't understand why it's the same logic after your change. Can you please explain?

Reply:
> I don't understand why it's the same logic after your change. Can you please explain?

It just mimicate <https://github.com/solana-labs/solana/blob/b7e7e4159c10cc249cdfc27d54b72218eebcea65/sdk/program/src/pubkey.rs#L474> It's semantically wrong to write `unsigned value <= 0` because this is not possible, even if the new code reach the limit of `UCHAR_MAX` the semantic is correct.